test/t8_forest/t8_gtest_forest_new_from_part_cmesh.cxx

```diff
@@ -0,0 +1,234 @@
+  t8_gloidx_t first = num_trees_before;
+  t8_gloidx_t second = num_trees_before + 1;
+  t8_gloidx_t third = num_trees_before + 2;
+  double vertices[9] = { (double) first, 0, 0, (double) second, 0, 0, (double) third, 0, 0 };
+  t8_cmesh_set_tree_class (cmesh, first, T8_ECLASS_LINE);
+  t8_cmesh_set_tree_vertices (cmesh, first, vertices, 2);
+  if (mpirank % 2 == 0) {
+    t8_cmesh_set_tree_class (cmesh, second, T8_ECLASS_LINE);
+    t8_cmesh_set_tree_vertices (cmesh, second, vertices + 3, 2);
+  }
+  t8_cmesh_set_partition_range (cmesh, face_knowledge, num_trees_before, num_trees_before + 1 - mpirank % 2);
+  t8_cmesh_commit (cmesh, comm);
+  int expected_local_trees = 2 - mpirank % 2;
+  int expected_global_trees = 3 * (mpisize / 2) + 2 * (mpisize % 2);
+  EXPECT_EQ (t8_cmesh_get_num_local_trees (cmesh), expected_local_trees);
+  EXPECT_EQ (t8_cmesh_get_num_trees (cmesh), expected_global_trees);
+  t8_scheme_cxx_t *scheme = t8_scheme_new_default_cxx ();
+  int level = 3;
+  t8_forest_t forest = t8_forest_new_uniform_with_cmesh_partition (cmesh, scheme, level, 0, comm);
+  EXPECT_EQ (t8_forest_get_local_num_elements (forest), expected_local_trees << level);
+  EXPECT_EQ (t8_forest_get_global_num_elements (forest), expected_global_trees << level);
+
+  t8_eclass_scheme_c *line_scheme = t8_forest_get_eclass_scheme (forest, T8_ECLASS_LINE);
+
+  for (t8_locidx_t itree = 0; itree < t8_forest_get_num_local_trees (forest); itree++) {
+    t8_tree_t tree = t8_forest_get_tree (forest, itree);
+    for (t8_locidx_t ielement = 0; ielement < t8_forest_get_tree_element_count (tree); ielement++) {
+      const t8_element_t *element = t8_forest_get_tree_element (tree, ielement);
+      EXPECT_EQ (line_scheme->t8_element_level (element), level);
+    }
+  }
+  t8_forest_unref (&forest);
+}
+
+TEST (t8_forest_new_from_partition, empty_test)
+{
+  /** TODO:
+   * hybrid
+   * parameterized above over eclass
+   * Look in cmesh examples, if there are any than can be constructed partitioned imbalanced.
+  */
+}
+
+TEST (t8_forest_new_from_partition, one_two_quad)
+{
+  /* Current process */
+  int rank, size;
+  sc_MPI_Comm_size (sc_MPI_COMM_WORLD, &size);
+  sc_MPI_Comm_rank (sc_MPI_COMM_WORLD, &rank);
+
+  /* Geometry */
+  int degree = 2;
+  std::vector<std::vector<double>> cells;
+  cells.push_back (std::vector<double> { -1, 1.2, 0,   0, 1.2,  0,   -1, 3,    0, 0, 3,    0,   -1, 2.1,
+                                         0,  0,   2.1, 0, -0.5, 1.2, 0,  -0.5, 3, 0, -0.5, 2.1, 0 });
+  cells.push_back (std::vector<double> { 0, 1.2, 0,   2, 1.5, 0,    0, 3, 0, 2, 3,   0,   0, 2.1,
+                                         0, 1.7, 2.1, 0, 1,   1.55, 0, 1, 3, 0, 0.9, 2.1, 0 });
+  cells.push_back (
+    std::vector<double> { 2, 1.5, 0, 4, 0, 0, 2, 3, 0, 4, 3, 0, 1.7, 2.1, 0, 4, 1.5, 0, 3, 1, 0, 3, 3, 0, 3, 2, 0 });
+
+  /* Create a partitioned cmesh */
+  t8_cmesh_t cmesh;
+  t8_cmesh_init (&cmesh);
+  if (size != 2)
```

Comment:
Take care of that SC_ABORT because the testsuite fails on single-procs this way.